examples/connectivity/compare_resting_state.py

```diff
@@ -0,0 +1,215 @@
+    data_filenames = dataset.func
+
+    print('First functional nifti image (4D) is at: %s' %
+          dataset.func[0])
+
+    # This is hacky and should be integrated in the nilearn API in a smooth way
+    # Warming up cache with masked images
+    print("[Example] Warming up cache")
+    decomposition_estimator = DecompositionEstimator(smoothing_fwhm=4.,
+                                                     memory=cache_dir,
+                                                     mask=mask,
+                                                     memory_level=2,
+                                                     verbose=10,
+                                                     n_jobs=n_jobs)
+    decomposition_estimator.fit(data_filenames, preload=True,
+                                temp_dir=temp_dir)
+    masker = decomposition_estimator.masker_
+
+    estimator_n_jobs = n_jobs if not parallel_exp else 1
+
+    alphas = [0.1]
+    estimators = []
+    for alpha in alphas:
+        sparse_pca = SparsePCA(n_components=n_components, mask=masker,
+                               memory="nilearn_cache", dict_init=dict_init,
+                               reduction_ratio=1,
+                               memory_level=2,
+                               alpha=alpha,
+                               batch_size=20,
+                               verbose=1,
+                               shuffle=True,
+                               random_state=0, l1_ratio=1,
+                               n_jobs=estimator_n_jobs,
+                               n_epochs=3)
+        estimators.append(sparse_pca)
+
+    with open(join(output, 'estimators'), mode='w+') as f:
+        pickle.dump(estimators, f)
+
+    exp_n_jobs = n_jobs if parallel_exp else 1
+
+    res = Parallel(n_jobs=exp_n_jobs, verbose=10)\
+        (delayed(fit_and_dump)(index,
+                               estimator,
+                               data_filenames,
+                               output)
+         for index, estimator
+         in enumerate(estimators))
+
+    components_filename, timings_list = zip(*res)
+    timings = np.zeros((len(estimators), 3))
+    for i, timing in enumerate(timings_list):
+        timings[i] = np.array(timing)
+    np.save(join(output, 'timings'), timings)
+
+    if len(estimators) > 1:
+        print("Performing alignment")
+        map_masker = MultiNiftiMasker(mask_img=masker.mask_img_,).fit()
+        components_list = align_list_with_last_nii(map_masker,
+                                                   components_filename)
+        comparison_dir = join(output, "comparison")
+        os.mkdir(comparison_dir)
+        Parallel(n_jobs=n_jobs)(
+            delayed(dump_nii_and_pdf)(i, components, comparison_dir)
+            for i, components in enumerate(components_list))
+
+if __name__ == '__main__':
+    t0 = time.time()
+    run_experiment(n_jobs=1, dataset='adhd')
+    time = time.time() - t0
+    print('Total_time : %f s' % time)
```

Comment:
Without going deep into details, the example that you propose is clearly much more sophisticated than those we already have. For instance, giving options  for many parameters is interesting but is no longer the 20 to 50 lines script that we had in most cases.
I think that skl has the concept of "real life examples", which are clearly different from the vanilla ones. We should probably follow this distinction and put your in a different place.